Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp

```diff
@@ -6,307 +6,264 @@
-            AzFramework::WindowSize newWindowSize;
-            AzFramework::WindowRequestBus::EventResult(newWindowSize, windowHandle, &AzFramework::WindowRequests::GetClientAreaSize);
-            AZ_Error("ResizeClientArea", newWindowSize.m_width == width && newWindowSize.m_height == height,
-                "Requested window resize to %ux%u but got %ux%u. This display resolution is too low or desktop scaling is too high.",
-                width, height, newWindowSize.m_width, newWindowSize.m_height);
-        }
-
-        bool SupportsToggleFullScreenOfDefaultWindow()
+        void Error(const AZStd::string& message)
         {
-            return AzFramework::NativeWindow::CanToggleFullScreenStateOfDefaultWindow();
-        }
+            auto operation = [message]()
+            {
+                AZ_Error("ScriptAutomation", false, "Script: %s", message.c_str());
+            };
 
-        void ToggleFullScreenOfDefaultWindow()
-        {
-            AzFramework::NativeWindow::ToggleFullScreenStateOfDefaultWindow();
+            ScriptAutomationInterface::Get()->QueueScriptOperation(AZStd::move(operation));
         }
 
-        AZ::IO::FixedMaxPath GetProfilingPath(bool resolvePath)
+        void Warning(const AZStd::string& message)
         {
-            AZ::IO::FixedMaxPath path("@user@");
-            if (resolvePath)
+            auto operation = [message]()
             {
-                if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
-                {
-                    path.clear();
-                    settingsRegistry->Get(path.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_ProjectUserPath);
-                }
-            }
-            path /= "scriptautomation/profiling";
-
-            return path.LexicallyNormal();
-        }
+                AZ_Warning("ScriptAutomation", false, "Script: %s", message.c_str());
+            };
 
-        AZ::IO::FixedMaxPath ResolvePath(const AZ::IO::PathView path)
-        {
-            AZStd::optional<AZ::IO::FixedMaxPath> resolvedPath = AZ::IO::FileIOBase::GetInstance()->ResolvePath(path);
-            return resolvedPath ? resolvedPath.value() : AZ::IO::FixedMaxPath{};
+            ScriptAutomationInterface::Get()->QueueScriptOperation(AZStd::move(operation));
         }
-    } // namespace Utils
 
-    namespace Bindings
-    {
-        void Print(const AZStd::string& message [[maybe_unused]])
+        void Print(const AZStd::string& message)
         {
-#ifndef _RELEASE //AZ_TracePrintf does nothing in release, ignore this call
-            auto func = [message]()
+            auto operation = [message]()
             {
                 AZ_TracePrintf("ScriptAutomation", "Script: %s\n", message.c_str());
             };
 
-            ScriptAutomationInterface::Get()->QueueScriptOperation(AZStd::move(func));
-#endif
+            ScriptAutomationInterface::Get()->QueueScriptOperation(AZStd::move(operation));
```

Comment:
AZ_TracePrintF, AZ_Warning, & AZ_Error are all nop's in release, please re-add the #ifndef's